### PR TITLE
WSTEAM-162: Enable Article BFF fetching on all services

### PR DESCRIPTION
**Pull request by shayneahchoon**
Resolves WSTEAM-162

**Overall change:**
- Enables fetching from the BFF for article pages for all services

**Code changes:**
- Updates fixture data for articles to match the new BFF format
- Updates tests to map to the new data format for the BFF response
- Removes the old fetch method in `getInitialData` to now use the BFF fetch mechanism

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/routes/index.test.jsx [hunk 1]
```diff
@@ -42,6 +43,10 @@ global.performance.getEntriesByName = jest.fn(() => []);
 // mock pages/index.js to return a non async page component
 jest.mock('../pages');
 
+const agent = { ca: 'ca', key: 'key' };
+const getAgent = jest.fn(() => agent);
+const fetchDataSpy = jest.spyOn(fetchPageData, 'default');
```
Comment (eagerterrier) [inline src/app/routes/index.test.jsx:48]: Not sure I understand this. If this makes me unqualified then disregard my CR

File: src/app/routes/index.test.jsx [hunk 2]
```diff
@@ -533,13 +550,17 @@ it('should route to and render a 404 error page', async () => {
 
 it('should render a 404 error page if a data fetch responds with a 404', async () => {
   const pathname = '/pidgin/articles/cwl08rd38p6o';
-  fetchMock.mock(`http://localhost${pathname}.json`, 404);
+  const bffFetchSpy = jest.spyOn(fetchPageData, 'default');
+
+  bffFetchSpy.mockRejectedValue({ message: 'Not found', status: 404 });
```
Comment (eagerterrier) [inline src/app/routes/index.test.jsx:555]: This is way beyond my comprehension now. You are a magician

**Review by eagerterrier — APPROVED** [review]
Wow. A lot of this was beyond my comprehension. You need to teach me some of this!!!

the bits I could understand (~80%) were great

File: src/app/routes/index.test.jsx (hunk 1 above)
Comment (amoore108) [reply src/app/routes/index.test.jsx:48]: Haha its all good!

The `agent` stuff is basically mocking up the return values from the `getAgent` function in our data fetcher. `getAgent` allows us to make use of the certificates configured on the server to authenticate our requests to FABL. Previously, the `.json` endpoints were open to the public, whereas FABL is behind a cert.

The `fetchDataSpy` uses Jests `spyOn` function to create a mock function of `fetchPageData`. This lets us then mock the return responses from `fetchPageData`, as seen in your comment below. The `'default'` part of the `spyOn` is just to tell Jest that `fetchPageData` is the default export.

File: src/app/routes/index.test.jsx (hunk 2 above)
Comment (amoore108) [reply src/app/routes/index.test.jsx:555]: From the comment above, the `spyOn` function lets us mock the return responses from whatever function we are 'spying' on. In this case, we are forcing a `404` using the `mockRejectedValue` function. Behind the scenes, this creates a `Promise.reject`, which forces the code in the BFF fetch to hit the `catch` part of our `try/catch`, and so returns an error response.

This needed changed from the old `.json` route as we aren't using that anymore for BFF fetching.

File: src/app/routes/utils/sharedDataTransformers/addIdsToBlocks/index.test.js [hunk 3]
```diff
@@ -7,194 +7,19 @@ jest.mock('uuid', () => ({
 
 describe('addIdsToBlocks rule', () => {
   it('should recursively add ids to blocks in article data', () => {
```
Comment (HarveyPeachey) [inline src/app/routes/utils/sharedDataTransformers/addIdsToBlocks/index.test.js]:
```suggestion
  it('should recursively add ids to blocks in cps asset data', () => {
```

**Review by HarveyPeachey — APPROVED** [review]

**Pull request merged**